distributed/tests/test_worker.py

```diff
@@ -663,6 +663,27 @@ async def test_close_on_disconnect(s, w):
         assert time() < start + 5
 
 
+@gen_cluster(nthreads=[("127.0.0.1", 1)])
+async def test_worker_close_closes_monitor(s, w):
```

Comment:
This seems pretty indirect from the actual thing we want the `close` method to do. Instead of testing that `close()` is called, it'd be good to have a more direct test that the gil monitor has been shutdown. 

Also, if we're able to centralize this logic in `Server`, I suspect we can avoid having two very similar tests for `Scheduler` and `Worker` in favor of a single test for `Server`

Reply:
Updated, looks much better than these two close sibling tests. :)